server/users.ts

```diff
@@ -522,22 +522,8 @@ export class User extends Chat.MessageContext {
 	}
 	can(permission: string, target: User | null = null, room: Room | BasicChatRoom | null = null): boolean {
 		if (this.hasSysopAccess()) return true;
-
-		const auth: Auth = room ? room.auth : Users.globalAuth;
-
-		let group = auth.get(this);
-		if (auth.has(this.id) && group === Auth.defaultSymbol()) {
-			group = 'whitelist' as GroupSymbol;
-		}
-		const targetGroup = target ? auth.get(target) : undefined;
-
-		const roomIsTemporary = room && !room.persist;
-		if (roomIsTemporary && group === this.group) {
-			const replaceGroup = Auth.getGroup(group).globalGroupInPersonalRoom;
-			if (replaceGroup) group = replaceGroup;
-		}
-
-		return Auth.hasPermission(group, permission, targetGroup, target === this);
+		if (!room) room = Rooms.global;
```

Comment:
:(

tbh I wrote #6885 just because I really didn't like this line.

Why did you write it, anyway? What is this intended to accomplish?

Reply:
This, as i recall, was intended to support changing global perms in the future.

Reply:
I'm also fairly certain it was also partially because I needed a room to be passed and didn't want to have it pass a specific room if it wasn't a room-only command, so I decided; two birds one stone.

Reply:
Global permissions should be associated with `Users.globalAuth` (support passing an `auth` object), and `room` should be nullable. That would be a better solution to your problems.

Reply:
This is not resolved. You should not consider `Rooms.global` a room.